_posts/2025-04-09-#31664.md

```diff
@@ -10,17 +10,120 @@ status: upcoming
+- [Issue #27995 by Sipa on improving fee estimation](https://github.com/bitcoin/bitcoin/issues/27995)
+- [Blog Post Discussing challenges of fee estimation](https://hackmd.io/@kEyqkad6QderjWKtcBF5Hg/cChallengies-with-estimating-transaction-fees)
+- [Delving Bitcoin Post: Mempool-Based Fee Estimation](https://delvingbitcoin.org/t/mempool-based-fee-estimation-on-bitcoin-core/703)
+- [Empirical Data Analysis of Fee Rate Forecasters for ASAP (next-block) Fee Estimation](https://delvingbitcoin.org/t/empirical-data-analysis-of-fee-rate-forecasters-for-asap-next-block-fee-estimation/1022)
+
+## Questions
+### **Conceptual & Approach**
 1. Did you review the PR? [Concept ACK, approach ACK, tested ACK, or NACK](https://github.com/bitcoin/bitcoin/blob/master/CONTRIBUTING.md#peer-review)? What was your review approach?
--->
+2. Why is the new system called a "Forecaster" and "ForecasterManager" rather than an "Estimator" and "Fee Estimation Manager"?
+3. Why is `CBlockPolicyEstimator` not modified to hold the mempool reference, similar to the approach in PR #12966 [`#12966`](https://github.com/bitcoin/bitcoin/pull/12966/) Why is the current approach chosen and why is it better? Hint see [`#28368`](https://github.com/bitcoin/bitcoin/pull/28368/)
+4. What are the trade-offs between the new architecture and a direct modification of `CBlockPolicyEstimator`?
+
+### **Code Review & Implementation**
+5. Why does [Commit 1e6ce06](https://github.com/bitcoin/bitcoin/pull/31664/commits/1e6ce06bf34eb3179f807efbddb0e9bca2d27f28) compare against only the high-priority estimate?
+6. What other methods might be useful in the `Forecaster` interface ([Commit a2e3326](https://github.com/bitcoin/bitcoin/pull/31664/commits/a2e33262b9f2fcef6b12e5a35fa736fc3a5c458d))?
+7. Why does [Commit 143a301](https://github.com/bitcoin/bitcoin/pull/31664/commits/143a301d3a3fc61e314b68308a47191543c75421) return the current height, and where is `nBestSeenHeight` set in the code?
+8. Why is it important to maintain monotonicity when iterating through the package fee rates in ([Commit 61e2842](https://github.com/bitcoin/bitcoin/pull/31664/commits/61e28422f3ff7dd6e1c840b9eac442c1484aa015))?
+9. Why were the 75th and 50th percentile fee rates chosen as `MempoolForecaster` fee rate estimate in ([Commit c7cdeaf](https://github.com/bitcoin/bitcoin/pull/31664/commits/c7cdeafd7045c794b289e2438cc60baeee7e6f69))?
+10. In what way do we also benefit from making `CBlockPolicyEstimator` a `shared_ptr` in ([Commit e8f5eb514](https://github.com/bitcoin/bitcoin/pull/31664/commits/e8f5eb51434f9b75b7c3f347fe073a9b60eb977c))?
+11. Why are mempool forecaster estimates cached for 30 seconds? Could a different duration be better ([Commit 5bd2220](https://github.com/bitcoin/bitcoin/pull/31664/commits/5bd2220422af02b20cb94dcd4c0e2c38e0284952))?
+12. Should caching and locking be managed within `MempoolForecaster` instead of `CachedMempoolEstimates`? Why is it declared mutable in `MempoolForecaster`? 
```

Comment:
What does "it" refer to? Maybe link to the declaration?

Reply:
Done